### PR TITLE
drivers: wifi: nrfwifi: Enable variable network configurations

### PR DESCRIPTION
**Pull request by D-Triveni**
Add a name to the existing Kconfig choice for better clarity. Add variable network configuration control into the driver, removing the need for per-sample settings.

### CODE REVIEW

**Review by sachinthegreen — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

**Comment by rlubos** [comment]
It seems that `nucleo_h723zg` + shield configuration was already on the edge when it comes to RAM (99.53% on main), with the massive net buf space it now overflows. Not sure if it's better to keep the defaults smaller, or choose different platform for testing.

File: subsys/net/ip/Kconfig [hunk 1]
```diff
@@ -792,7 +792,7 @@ config NET_BUF_TX_COUNT
 	  Each data buffer will occupy CONFIG_NET_BUF_DATA_SIZE + smallish
 	  header (sizeof(struct net_buf)) amount of data.
 
-choice
+choice NET_PKT_DATA_ALLOC_TYPE
```
Comment (krish2718) [inline subsys/net/ip/Kconfig:795]: This should be in a separate commit as a pre-requisite

File: drivers/wifi/nrfwifi/Kconfig.nrfwifi [hunk 2]
```diff
@@ -555,6 +555,18 @@ config NET_TC_TX_COUNT
...
+	default NET_BUF_VARIABLE_DATA_SIZE
+endchoice
+
+if NET_BUF_VARIABLE_DATA_SIZE
+config NET_PKT_BUF_RX_DATA_POOL_SIZE
+	default 32000
+
+config NET_PKT_BUF_TX_DATA_POOL_SIZE
+	default 55000
+endif
```
Comment (krish2718) [inline drivers/wifi/nrfwifi/Kconfig.nrfwifi:568]: Please add some comments as to how we have arrived at these numbers and if there is specific config or results (throughput).

**Review by sachinthegreen — DISMISSED** [review]

File: drivers/wifi/nrfwifi/Kconfig.nrfwifi [hunk 3]
```diff
@@ -555,6 +555,23 @@ config NET_TC_TX_COUNT
...
```
Comment (krish2718) [inline drivers/wifi/nrfwifi/Kconfig.nrfwifi:573]: should we protect this with a Kconfig? with default still being variable buffers?

**Review by sachinthegreen — DISMISSED** [review]

**Review by jukkar — DISMISSED** [review]

File: drivers/wifi/nrfwifi/Kconfig.nrfwifi [hunk 4]
```diff
@@ -570,6 +570,39 @@ config NET_TC_TX_COUNT
...
+	bool "Variable Data Size"
+	help
+	  Use variable data sizes for packet buffers.
+
+config NRF_WIFI_FIXED_DATA_SIZE
+	bool "Fixed Data Size"
+	help
+	  Use fixed data sizes for packet buffers.
+
+endchoice
```
Comment (krish2718) [inline drivers/wifi/nrfwifi/Kconfig.nrfwifi:587]:
```suggestion
choice NRF_WIFI_NET_BUF_DATA_ALLOC_TYPE
	prompt "Network buffer type"
	default NRF_WIFI_VARIABLE_DATA_SIZE

config NRF_WIFI_NET_BUF_VARIABLE_SIZE
	bool "Variable Data Size"
	help
	  Use variable network buffer  sizes for packet buffers.

config NRF_WIFI_NET_BUG_FIXED_SIZE
	bool "Fixed Data Size"
	help
	  Use fixed data sizes for packet buffers.

endchoice
```

**Review by sachinthegreen — DISMISSED** [review]

File: drivers/wifi/nrfwifi/Kconfig.nrfwifi [hunk 5]
```diff
@@ -570,6 +570,39 @@ config NET_TC_TX_COUNT
 
 endif # NETWORKING
 
+choice NRF_WIFI_NET_BUF_DATA_ALLOC_TYPE
+	prompt "Network buffer type"
```
Comment (rado17) [inline drivers/wifi/nrfwifi/Kconfig.nrfwifi:574]: If this overrides the `ALLOC_TYPE` as defined under `net`, we should mention that in prompt.

**Review by rado17 — APPROVED** [review]

**Review by sachinthegreen — DISMISSED** [review]

**Review by VivekUppunda — APPROVED** [review]

**Review by krish2718 — DISMISSED** [review]

**Comment by krish2718** [comment]
Remove adding another abstraction over networking stack, and also removed overrides for highest throughput as they impact all samples/applications RAM usage.

**Review by jukkar — APPROVED** [review]

**Comment by kartben** [comment]
@D-Triveni this needs to be rebased please
